datafusion/core/src/physical_plan/udaf.rs

```diff
@@ -70,6 +70,11 @@ impl AggregateFunctionExpr {
     pub fn fun(&self) -> &AggregateUDF {
         &self.fun
     }
+
+    /// Returns true if this can support sliding accumulators
+    pub fn retractable(&self) -> Result<bool> {
+        Ok((self.fun.accumulator)(&self.data_type)?.supports_retract_batch())
```

Comment:
it is non ideal that we need to instanatiate the accumulator just to check if it supports retract batch, but that is consistent with the window function API so I think that is ok

Reply:
Maybe we can fix this and its window function counterpart in a refactor. We discussed this instantiation thing in a meeting recently and we are not super happy with it either. Let's get the basics in and then we can work on the rough edges.

Reply:
Another other formulation I tried looked like the following. The upside is it avoided the instantiation, but the downside is that it an API change and is not consistent with the window functions

```rust
pub struct AggregateUDF {
    /// name
    pub name: String,
    /// Signature (input arguments)
    pub signature: Signature,
    /// Return type
    pub return_type: ReturnTypeFunction,
    /// Return an accumulator without retract batch
    pub accumulator: AccumulatorFunctionImplementation,
    /// Return an accumulator with retract batch
    pub retractable_accumulator: Option<AccumulatorFunctionImplementation>,  <---- this is added
    /// the accumulator's state's description as a function of the return type
    pub state_type: StateTypeFunction,
}
```